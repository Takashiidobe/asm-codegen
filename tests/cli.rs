use std::process::Stdio;

use assert_cmd::cargo::CommandCargoExt as _;
use insta::glob;
use insta_cmd::{assert_cmd_snapshot, Command};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct TestOutput {
    status: i32,
    stdout: Vec<String>,
    stderr: Vec<String>,
}

#[test]
fn gen_bin() {
    glob!("input", "**/*.my", |path| {
        let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();

        let ps_child = cmd
            .arg(path)
            .stdout(Stdio::piped())
            .stdout(Stdio::piped())
            .spawn()
            .expect("Could not create assembly");

        let out_name = format!("{}.out", path.display());

        Command::new("cc")
            .arg("-o")
            .arg(&out_name)
            .arg("-xassembler")
            .arg("-")
            .stdin(Stdio::from(ps_child.stdout.unwrap()))
            .spawn()
            .expect("Could not assemble binary");

        assert_cmd_snapshot!(Command::new(&out_name));
    });
}
