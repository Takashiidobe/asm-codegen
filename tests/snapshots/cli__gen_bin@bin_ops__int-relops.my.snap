---
source: tests/cli.rs
info:
  program: int-relops.my.out
  args: []
input_file: tests/input/bin_ops/int-relops.my
---
success: true
exit_code: 0
----- stdout -----
should be all true
true
true
true
true
true
true
true
true
should be all false
false
false
false
false
false
false
false
false

----- stderr -----
